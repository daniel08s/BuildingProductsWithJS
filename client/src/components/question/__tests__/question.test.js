/* global test expect shallow React */
// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import QuestionWrapper, {Question} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '1',
  login: 'testuser',
};

const question = {
  owner: user,
  text: 'Test question',
  answers: [{answer: 'Test answer'}],
};

test('# QuestionWrapper', () => {
  const store = mockStore({auth: {user}});

  const wrapper = shallow(<QuestionWrapper store={store} question={question} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Question', () => {
  const answer = 'Test answer';
  const updatedText = 'New test answer';
  const onAnswer = ({question: q, answer: a}) => {
    expect(q).toEqual(question);
    expect(a).toBe(answer);
  };
  const deleteQuestion = q => expect(q).toEqual(question);
  const updateQuestion = ({text}) => expect(text).toBe(updatedText);
  const component = (
    <Question
      user={user}
      question={question}
      onAnswer={onAnswer}
      deleteQuestion={deleteQuestion}
      updateQuestion={updateQuestion}
    />
  );
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // test interaction
  const app = mount(component);

  /* Test Answer */
  // set answer
  app.find('#answerInput').getDOMNode().value = answer;

  // click answer button
  app.find('#answerBtn').simulate('click');

  /* Test Delete */
  app.find('#deleteBtn').simulate('click');

  /* Test Update */
  // enable editing
  app.find('#editBtn').simulate('click');

  // update question
  app.find('#questionText').getDOMNode().value = updatedText;

  // click update button
  app.find('#updateBtn').simulate('click');
});
