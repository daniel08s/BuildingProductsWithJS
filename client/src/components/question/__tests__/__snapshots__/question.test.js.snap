// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Question 1`] = `
<div
  className="panel panel-default text-left"
>
  <div
    className="panel-heading"
  >
    <span>
      <button
        className="btn btn-link"
        id="deleteBtn"
        onClick={[Function]}
        type="submit"
      >
        <span
          className="glyphicon glyphicon-trash"
          style={
            Object {
              "color": "#800000",
            }
          }
        />
      </button>
      <button
        className="btn btn-link"
        id="editBtn"
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-pencil"
          style={
            Object {
              "color": "#800000",
            }
          }
        />
      </button>
    </span>
    Test question
    <div
      className="pull-right"
    >
      Created by: 
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/profile/1"
      >
        testuser
      </Link>
    </div>
  </div>
  <div
    className="panel-body"
  >
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        Test answer
      </li>
    </ul>
  </div>
  <div
    className="panel-footer"
  >
    <form
      className="form-horizontal"
    >
      <div
        className="col-sm-10"
      >
        <input
          className="form-control"
          id="answerInput"
          placeholder="Enter your answer..."
          type="text"
        />
      </div>
      <button
        className="btn btn-default"
        id="answerBtn"
        onClick={[Function]}
        type="submit"
      >
        Answer
      </button>
    </form>
  </div>
</div>
`;

exports[`# QuestionWrapper 1`] = `
<Question
  deleteQuestion={[Function]}
  question={
    Object {
      "answers": Array [
        Object {
          "answer": "Test answer",
        },
      ],
      "owner": Object {
        "id": "1",
        "login": "testuser",
      },
      "text": "Test question",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  updateQuestion={[Function]}
  user={
    Object {
      "id": "1",
      "login": "testuser",
    }
  }
/>
`;
